sklearn/utils/estimator_checks.py

```diff
@@ -395,57 +402,138 @@ def _yield_all_checks(estimator, legacy: bool):
         yield check_fit_non_negative
 
 
-def _maybe_mark_xfail(estimator, check, pytest):
-    # Mark (estimator, check) pairs as XFAIL if needed (see conditions in
-    # _should_be_skipped_or_marked())
-    # This is similar to _maybe_skip(), but this one is used by
-    # @parametrize_with_checks() instead of check_estimator()
+def _check_name(check):
+    return check.func.__name__ if isinstance(check, partial) else check.__name__
+
 
-    should_be_marked, reason = _should_be_skipped_or_marked(estimator, check)
-    if not should_be_marked:
+def _maybe_mark(
```

Comment:
we had two functions, one marking them as `xfail` with pytest, another would wrap the test with something that would simply raise `SkipTest`. Those two are now merged into this function.